azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/transport/symmetric_key_transport.py

```diff
@@ -0,0 +1,24 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class SymmetricKeyTransport(object):
+    """
+    A super class of transports that use shared access signature token to connect.
+    """
+
+    def __init__(self, symmetric_key_security_provider):
+        self.security_provider = symmetric_key_security_provider
```

Comment:
>security_provider [](start = 13, length = 17)

same here - should this be underscored/private?

Reply:
As per my comment on the sk_security_provider, I don't really see a reason this has to be private by convention.

---
In reply to: [265325942](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265325942) [](ancestors = 265325942)

Reply:
because public means it becomes a public API that we have to document for the users and that we have to support. trust me on this, you want everything private by default and open things up only if you really need to. all these objects are going to be public, you don't want customer using these public fields to mess with things.

---
In reply to: [265328937](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265328937) [](ancestors = 265328937,265325942)

Reply:
Why would there be a public API for transport, which isn't part of the exposed API to begin with?

---
In reply to: [265330521](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265330521) [](ancestors = 265330521,265328937,265325942)

Reply:
Changed to private as of now as I don't quite remember why I did public.

---
In reply to: [265331363](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265331363) [](ancestors = 265331363,265330521,265328937,265325942)